src/COWShedResolver.sol

```diff
@@ -43,21 +43,31 @@ abstract contract COWShedResolver is INameResolver, IAddrResolver {
         return LibString.fromSmallString(baseNameSmallString);
     }
 
+    function supportsInterface(bytes4 interfaceId) external pure returns (bool) {
+        return interfaceId == IAddrResolver.addr.selector || interfaceId == INameResolver.name.selector;
```

Comment:
```suggestion
        return interfaceId == type(IAddrResolver).interfaceId || interfaceId == type(INameResolver).interfaceId || type(IERC165).interfaceId;
```
nit: use `interfaceId` property of `type(Interface)`, and include support for `IERC165`.